nilearn/reporting/glm_reporter.py

```diff
@@ -967,3 +973,284 @@ def _dataframe_to_html(df, precision, **kwargs):
         html_table = df.to_html(**kwargs)
     html_table = html_table.replace('border="1" ', '')
     return html_table
+
+
+def _clean_contrast_name(contrast_name):
+    """Remove non-alphanumeric characters from a contrast name.
+
+    BIDS filenames, in which the contrast name will appear as a
+    contrast-<name> key/value pair, must be alphanumeric strings.
+
+    Parameters
+    ----------
+    contrast_name : :obj:`str`
+        Contrast name to clean.
+
+    Returns
+    -------
+    new_name : :obj:`str`
+        Contrast name with non-alphanumeric characters removed.
+    """
+    new_name = ''.join(ch for ch in contrast_name if ch.isalnum())
+    if new_name != contrast_name:
+        warnings.warn(
+            'Contrast name "{}" changed to "{}"'.format(
+                contrast_name,
+                new_name
+            )
+        )
+    return new_name
+
+
+def save_glm_results(model, contrasts, out_dir='.', prefix=None):
+    """Save GLM results to BIDS-like files.
+
+    Parameters
+    ----------
+    model : :obj:`nilearn.glm.first_level.FirstLevelModel` or
+            :obj:`nilearn.glm.second_level.SecondLevelModel`
+        First- or second-level model from which to save outputs.
+    contrasts : :obj:`dict` of :obj:`str`-:obj:`numpy.ndarray` pairs
+        A dictionary containing contrasts. Keys are contrast names,
+        while values are arrays containing contrast weights.
+        The arrays may be 1D (for t-contrasts) or 2D (for F-contrasts).
+    out_dir : :obj:`str`, optional
+        Output directory for files. Default is current working directory.
+    prefix : :obj:`str` or None, optional
+        String to prepend to generated filenames.
+        If a string is provided, "_" will be added to the end.
+        Default is None.
+
+    Warning
+    -------
+    The files generated by this function are a best approximation of
+    appropriate for a GLM-based BIDS derivative.
```

Comment:
Is it possible to add a ref to a given spec ?

Reply:
Outputs aren't written up in a spec yet, but the idea was brought up recently in https://github.com/poldracklab/fitlins/issues/299.

Reply:
I've started working on translating the `fitlins` rules into a BEP that we can reference: https://docs.google.com/document/d/1OxKhhEctlQg6nmw1L3vw9Mm0K9-KkTpVKZuNqYnlYEM/edit?usp=sharing